apps/backend/user/src/confirmEmail.ts

```diff
@@ -1,14 +1,42 @@
 import { Request, Response } from "express";
 import { IDatabase } from "pg-promise";
+import { verify, JwtPayload } from "jsonwebtoken";
 
 const confirmEmail = async (
   req: Request,
   res: Response,
   db: IDatabase<object>,
 ) => {
   const code = req.body.code;
-  console.log(code); // here so code is not an unused variable
-  const userId = 2; // get from code (jwt token)
+
+  if (!code) {
+    return res.status(400).json({ error: "Code is required" });
+  }
+
+  const JWT_PUBLIC_KEY =
+    process.env.JWT_PUBLIC_KEY ||
+    (() => {
+      throw new Error("JWT_PUBLIC_KEY is not set");
+    })();
+
+  let decoded: JwtPayload & { user_id: number };
+
+  try {
+    decoded = verify(code, JWT_PUBLIC_KEY, {
+      algorithms: ["RS256"],
+    }) as JwtPayload & { user_id: number };
+  } catch (error) {
+    console.error(error);
+    return res.status(401).json({ error: "Invalid code" });
+  }
+
+  // The userId is currently stored incorrectly in the jwt as structured: userId = { user_id: number }
+  // This should be updated at some point
+  const userId = decoded?.userId?.user_id;
+
+  if (!userId) {
+    return res.status(400).json({ error: "Invalid user ID in code" });
+  }
```

Comment:
Why are we doing JWT things here? This endpoint should be called from the frontend with the auth cookie (and then the auth Middleware will magically handle this case if you restrict the token to this route only)

Reply:
Because the confirm email page is coming from the link sent in an email, so a cookie can’t be sent